packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -186,6 +187,7 @@ function resetComponentStateWhenRemoved(vm: VM) {
 // old vnode.children is removed from the DOM.
 export function removeVM(vm: VM) {
     if (process.env.NODE_ENV !== 'production') {
+        removeHotVM(vm);
```

Comment:
this is to avoid leaking VM instances for components that were removed, but their template still available for new instances. this just complements `addHotVM`